alibi_detect/cd/base.py

```diff
@@ -317,13 +381,23 @@ def __init__(
         if p_val is None:
             logger.warning('No p-value set for the drift threshold. Need to set it to detect data drift.')
 
-        # optionally already preprocess reference data
-        self.p_val = p_val
-        if preprocess_x_ref and isinstance(preprocess_fn, Callable):  # type: ignore[arg-type]
+        # x_ref preprocessing logic
```

Comment:
I agree with you that it would certainly be nice to factor out. But, I didn't want to do as part of this PR as it's a change to the core of the detectors and not really related to the config functionality (it only looks as if I've added stuff above as have had to rework the logic a bit due to the kwarg changes).

Also think above is motivation for adding a drift detector base class i.e. `BaseDrift(BaseDetector)` which `BaseUnivariateDrift` would inherit from, rather than adding lots of Mixin's...